apps/api-documenter/src/documenters/MarkdownDocumenter.ts

```diff
@@ -0,0 +1,694 @@
+              ])
+            );
+          }
+
+          this._writeApiItemPage(apiMember);
+          break;
+        }
+
+      }
+    }
+
+    if (eventsTable.rows.length > 0) {
+      output.appendNode(new DocHeading({ configuration: this._tsdocConfiguration, title: 'Events' }));
+      output.appendNode(eventsTable);
+    }
+
+    if (propertiesTable.rows.length > 0) {
+      output.appendNode(new DocHeading({ configuration: this._tsdocConfiguration, title: 'Properties' }));
+      output.appendNode(propertiesTable);
+    }
+
+    if (methodsTable.rows.length > 0) {
+      output.appendNode(new DocHeading({ configuration: this._tsdocConfiguration, title: 'Methods' }));
+      output.appendNode(methodsTable);
+    }
+  }
+
+  /**
+   * GENERATE PAGE: FUNCTION-LIKE
+   */
+  private _writeFunctionLikeTables(output: DocSection, apiFunctionLike: ApiFunctionLikeMixin): void {
+    const configuration: TSDocConfiguration = this._tsdocConfiguration;
+
+    const parametersTable: DocTable = new DocTable({ configuration,
+      headerTitles: [ 'Parameter', 'Type', 'Description' ]
+    });
+
+    for (const apiParameter of apiFunctionLike.parameters) {
+      const parameterDescription: DocSection = new DocSection({ configuration } );
+      if (apiParameter.tsdocParamBlock) {
+        this._appendSection(parameterDescription, apiParameter.tsdocParamBlock.content);
+      }
+
+      parametersTable.addRow(
+        new DocTableRow({ configuration }, [
+          new DocTableCell({configuration}, [
+            new DocParagraph({ configuration }, [
+              new DocPlainText({ configuration, text: apiParameter.name })
+            ])
+          ]),
+          new DocTableCell({configuration}, [
+            new DocParagraph({ configuration }, [
+              new DocCodeSpan({ configuration, code: apiParameter.resultTypeSignature })
+            ])
+          ]),
+          new DocTableCell({configuration}, parameterDescription.nodes)
+        ])
+      );
+    }
+
+    if (parametersTable.rows.length > 0) {
+      output.appendNode(new DocHeading({ configuration: this._tsdocConfiguration, title: 'Parameters' }));
+      output.appendNode(parametersTable);
+    }
+
+    if (ApiResultTypeMixin.isBaseClassOf(apiFunctionLike)) {
+
+      output.appendNode(
+        new DocParagraph({ configuration }, [
+          new DocEmphasisSpan({ configuration, bold: true}, [
+            new DocPlainText({ configuration, text: 'Returns:' })
+          ])
+        ])
+      );
+
+      output.appendNode(
+        new DocParagraph({ configuration }, [
+          new DocCodeSpan({ configuration, code: apiFunctionLike.resultTypeSignature })
+        ])
+      );
+
+      if (apiFunctionLike instanceof ApiDocumentedItem) {
+        if (apiFunctionLike.tsdocComment) {
+          if (apiFunctionLike.tsdocComment.returnsBlock) {
+            this._appendSection(output, apiFunctionLike.tsdocComment.returnsBlock.content);
```

Comment:
Honestly just curious, are nested if statements a personal preference? Instead of using something like `if(apiFunctionLike.tsdocComment && apiFunctionLike.tsdocComment.returnsBlock) { ... }`

Reply:
@natalieethell I used them here because each "if" statement narrows the type of the object, adding more stuff to its IntelliSense, which helped when writing the code.  I should probably collapse at least one of them to improve readability.

---
In reply to: [236366973](https://github.com/Microsoft/web-build-tools/pull/952#discussion_r236366973) [](ancestors = 236366973)